src/Microsoft.DotNet.ProjectJsonMigration/transforms/AddPropertyTransform.cs

```diff
@@ -32,7 +32,13 @@ public AddPropertyTransform(string propertyName, Func<T, string> propertyValueFu
 
         public AddPropertyTransform<T> WithMSBuildCondition(string condition)
         {
-            _msbuildCondition = condition;
+            _msbuildConditionFunc = source => condition;
```

Comment:
Is null or empty valid? This kind of indirection can be pretty hard to debug if we accept bad data to be used in a deferred call.